### PR TITLE
Port shimming-toolbox documentation

### LINKED ISSUE #38: Move ReadTheDocs code from shimming-toolbox
Most of the existing documentation for shimming-toolbox is not MATLAB-specific, so it's a good time to port over the documentation into this project to keep project development centralized. This will ensure that documentation can be updated alongside feature implementations.

The tasks involved are:

- [x] move RTD code into this repo
- [x] use Miniconda instructions from AxonDeepSeg project
- [x] ~~CNAME reassignment~~ outside this repo
- [x] ~~adding a RTD build step to Travis and/or GitHub Actions~~ (RTD can be configured to track the relevant repo without any need for automation on Travis or GH Actions)

### PR DESCRIPTION
**Pull request by jidicula**
The RTD .md files are copied over from [shimming-toolbox](shimming-toolbox/shimming-toolbox) and converted using pandoc into .rst files. RTD build steps are adapted from [AxonDeepSeg](neuropoly/axondeepseg).

API documentation autogeneration has not yet been included in the RTD config.

Resolves #38 
Related to #2

### CODE REVIEW

File: docs/1_overview/1_introduction.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Introduction
```
Comment (jcohenadad) [inline docs/1_overview/1_introduction.md:1]: i suggest we make it a 100% rst format, which has better compatibility with sphinx

**Comment by jidicula** [comment]
^ I mentioned the wrong issue in the first commit message and had to amend it to prevent GitHub from autoclosing the wrong PR

**Comment by po09i** [comment]
Another relevant issue in this repo: #2
We had discussions on the [shimming-toolbox.org repo](https://github.com/shimming-toolbox/shimming-toolbox.org) repo about where to host the doc. We had originally settled on mkdocs, gh-pages and azure-pipelines to be able to run matlab for generating the documentation. Now that we don't need matlab, azurepipelines is not required. 

So I would go with Sphinx for generating the doc, RTD for hosting and either travis and/or github actions for automating the build.

**Comment by jcohenadad** [comment]
> So I would go with Sphinx for generating the doc, RTD for hosting and either travis and/or github actions for automating the build.

actually, RTD does automate the build, so no need for travis or GH actions. RTD receives notifs from GH for every event (push, pr, etc.)

**Comment by jidicula** [comment]
Great! I'll just use pandoc for converting the existing relevant (not MATLAB-specific) .md files to .rst.

**Comment by jidicula** [comment]
I think all that's left is the CNAME reassignment.

**Comment by gab-berest** [comment]
> 
> 
> I think all that's left is the CNAME reassignment and API documentation generation, which I'm not sure how to do.

I don't know exactly what we want for the API documentation, but if it is to show the API/CLI commands that can be used, it would be to parse the click entries and function params and returns in the separate files to generate the documentation. It could be done with a Python script. There's also the parser and docstring-parser module that we could investigate into. I also found this [https://tomassetti.me/parsing-in-python/#parserGenerators](url) that could be usefull for automatic generation.

**Comment by jcohenadad** [comment]
> I think all that's left is the CNAME reassignment and API documentation generation, which I'm not sure how to do.

we could leave the API generation for another PR. i'm afraid we will accumulate conflicts if we wait to long, eg. https://github.com/shimming-toolbox/shimming-toolbox-py/pull/43/files#diff-04c6e90faac2675aa89e2176d2eec7d8

**Comment by jidicula** [comment]
I figured it out – I've added automodule directives for each module in a separate section for API documentation. However, I'm getting a doc build error from the numerical_model module:

```
/Users/johanan/neuropoly/shimming-toolbox-py/shimmingtoolbox/simulate/numerical_model.py:docstring of shimmingtoolbox.simulate.numerical_model.NumericalModel:11: WARNING: Unexpected indentation.
/Users/johanan/neuropoly/shimming-toolbox-py/shimmingtoolbox/simulate/numerical_model.py:docstring of shimmingtoolbox.simulate.numerical_model.NumericalModel:12: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/johanan/neuropoly/shimming-toolbox-py/shimmingtoolbox/simulate/numerical_model.py:docstring of shimmingtoolbox.simulate.numerical_model.NumericalModel.save:10: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/johanan/neuropoly/shimming-toolbox-py/shimmingtoolbox/simulate/numerical_model.py:docstring of shimmingtoolbox.simulate.numerical_model.NumericalModel.save:12: WARNING: Unexpected indentation.
/Users/johanan/neuropoly/shimming-toolbox-py/shimmingtoolbox/simulate/numerical_model.py:docstring of shimmingtoolbox.simulate.numerical_model.NumericalModel.simulate_measurement:10: WARNING: Unexpected indentation.
/Users/johanan/neuropoly/shimming-toolbox-py/shimmingtoolbox/simulate/numerical_model.py:docstring of shimmingtoolbox.simulate.numerical_model.NumericalModel.simulate_measurement:11: WARNING: Block quote ends without a blank line; unexpected unindent.
```

I think there's an issue with the docstring formatting, but I think those fixes might be out of scope for this PR.

**Comment by jcohenadad** [comment]
> I think there's an issue with the docstring formatting in that class, but I think those fixes might be out of scope for this PR.

agreed 👍

**Review by jcohenadad — COMMENTED** [review]
i would suggest to call this project `shimming-toolbox` instead of `shimming-toolbox-py`, given that in the short term we will officially switch to the python implementation.

**Comment by jcohenadad** [comment]
@jidicula i'm seeing this from RTD:
Screen Shot 2020-07-30 at 11 38 56 AM

i'll configure the repos so it can send hooks to RTD build

**Comment by jcohenadad** [comment]
ok, the webhook is working, however the build failed: 
https://readthedocs.org/projects/shimming-toolbox-py/builds/11562434/

**Comment by jidicula** [comment]
> ok, the webhook is working, however the build failed:
> https://readthedocs.org/projects/shimming-toolbox-py/builds/11562434/

It was a tab issue – switched to spaces to match the YAML spec, so the build should be able to proceed.

**Comment by jcohenadad** [comment]
yay 🎉

---> https://shimming-toolbox-py.readthedocs.io/en/issue-38/

**Review by jcohenadad — APPROVED** [review]
looks good for a first iteration! thanks 🙏

**Comment by jcohenadad** [comment]
@jidicula i let you merge in case you want to fix minor stuff

**Comment by jidicula** [comment]
Doesn't look the API autogeneration is working on RTD 😔  – works fine for me locally

RTD:
image

Local build:
image


RTD warnings:
```
WARNING: autodoc: failed to import module 'download' from module 'shimmingtoolbox'; the following exception was raised:
No module named 'shimmingtoolbox'
WARNING: autodoc: failed to import module 'utils' from module 'shimmingtoolbox'; the following exception was raised:
No module named 'shimmingtoolbox'
WARNING: autodoc: failed to import module 'cli.b0map' from module 'shimmingtoolbox'; the following exception was raised:
No module named 'shimmingtoolbox'
WARNING: autodoc: failed to import module 'cli.download_data' from module 'shimmingtoolbox'; the following exception was raised:
No module named 'shimmingtoolbox'
WARNING: autodoc: failed to import module 'cli.referencemaps' from module 'shimmingtoolbox'; the following exception was raised:
No module named 'shimmingtoolbox'
WARNING: autodoc: failed to import module 'simulate.numerical_model' from module 'shimmingtoolbox'; the following exception was raised:
No module named 'shimmingtoolbox'
```

**Comment by jcohenadad** [comment]
> Doesn't look the API autogeneration is working on RTD 😔 – works fine for me locally

maybe check how we do it in https://ivadomed.org/en/latest/

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+# -- Path setup --------------------------------------------------------------
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+import os
+import sys
+
```
Comment (jcohenadad) [inline docs/source/conf.py:15]: you probably need to import shimmingtoolbox here to fix https://github.com/shimming-toolbox/shimming-toolbox-py/pull/41#issuecomment-666517789

Comment (jidicula) [reply docs/source/conf.py:15]: 🤦‍♂️ I had missed that earlier, thanks!

**Pull request merged**